`globalThis.crypto` type definition is broken since `"@types/bun": "1.2.7"`

### What version of Bun is running?

1.2.9+9a329c04c

### What platform is your computer?

Linux 6.13.8-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

```ts
const bytes = new Uint8Array(20);
globalThis.crypto.getRandomValues(bytes);
```

The above code has no type errors in `@types/bun` 1.2.6, but in versions 1.2.7~1.2.9 it results in the following error:
```sh
error TS2339: Property 'getRandomValues' does not exist on type 'Crypto'.
```

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Additional information

_No response_

--- a/packages/bun-types/bun.d.ts
+++ b/packages/bun-types/bun.d.ts
@@ -16,7 +16,9 @@
 declare module "bun" {
   type DistributedOmit<T, K extends PropertyKey> = T extends T ? Omit<T, K> : never;
   type PathLike = string | NodeJS.TypedArray | ArrayBufferLike | URL;
-  type ArrayBufferView = NodeJS.TypedArray | DataView;
+  type ArrayBufferView<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> =
+    | NodeJS.TypedArray<TArrayBuffer>
+    | DataView<TArrayBuffer>;
   type BufferSource = NodeJS.TypedArray | DataView | ArrayBufferLike;
   type StringOrBuffer = string | NodeJS.TypedArray | ArrayBufferLike;
   type XMLHttpRequestBodyInit = Blob | BufferSource | string | FormData | Iterable<Uint8Array>;

--- a/packages/bun-types/globals.d.ts
+++ b/packages/bun-types/globals.d.ts
@@ -1,9 +1,9 @@
 declare module "bun" {
   namespace __internal {
-    type NodeWorkerThreadsWorker = import("worker_threads").Worker;
+    type NodeWorkerThreadsWorker = import("node:worker_threads").Worker;
     type LibWorkerOrBunWorker = Bun.__internal.UseLibDomIfAvailable<"Worker", Bun.Worker>;
 
-    type NodePerfHooksPerformance = import("perf_hooks").Performance;
+    type NodePerfHooksPerformance = import("node:perf_hooks").Performance;
     type LibPerformanceOrNodePerfHooksPerformance = Bun.__internal.UseLibDomIfAvailable<
       "Performance",
       NodePerfHooksPerformance
@@ -12,19 +12,23 @@ declare module "bun" {
     type NodeCryptoWebcryptoSubtleCrypto = import("crypto").webcrypto.SubtleCrypto;
     type NodeCryptoWebcryptoCryptoKey = import("crypto").webcrypto.CryptoKey;
 
-    type LibEmptyOrNodeUtilTextEncoder = LibDomIsLoaded extends true ? {} : import("util").TextEncoder;
+    type LibEmptyOrNodeUtilTextEncoder = LibDomIsLoaded extends true ? {} : import("node:util").TextEncoder;
 
-    type LibEmptyOrNodeUtilTextDecoder = LibDomIsLoaded extends true ? {} : import("util").TextDecoder;
+    type LibEmptyOrNodeUtilTextDecoder = LibDomIsLoaded extends true ? {} : import("node:util").TextDecoder;
 
-    type LibEmptyOrNodeReadableStream<T> = LibDomIsLoaded extends true ? {} : import("stream/web").ReadableStream<T>;
+    type LibEmptyOrNodeReadableStream<T> = LibDomIsLoaded extends true
+      ? {}
+      : import("node:stream/web").ReadableStream<T>;
 
-    type LibEmptyOrNodeWritableStream<T> = LibDomIsLoaded extends true ? {} : import("stream/web").WritableStream<T>;
+    type LibEmptyOrNodeWritableStream<T> = LibDomIsLoaded extends true
+      ? {}
+      : import("node:stream/web").WritableStream<T>;
 
     type LibEmptyOrNodeTransformStream<I, O> = LibDomIsLoaded extends true
       ? {}
-      : import("stream/web").TransformStream<I, O>;
+      : import("node:stream/web").TransformStream<I, O>;
 
-    type LibEmptyOrNodeMessagePort = LibDomIsLoaded extends true ? {} : import("worker_threads").MessagePort;
+    type LibEmptyOrNodeMessagePort = LibDomIsLoaded extends true ? {} : import("node:worker_threads").MessagePort;
   }
 }
 
@@ -71,12 +75,23 @@ declare var Worker: Bun.__internal.UseLibDomIfAvailable<
  */
 declare var WebSocket: Bun.__internal.UseLibDomIfAvailable<"WebSocket", typeof import("ws").WebSocket>;
 
-interface Crypto {}
+interface Crypto {
+  readonly subtle: SubtleCrypto;
+
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Crypto/getRandomValues) */
+  getRandomValues<T extends ArrayBufferView | null>(array: T): T;
+
+  /**
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Crypto/randomUUID)
+   */
+  randomUUID(): `${string}-${string}-${string}-${string}-${string}`;
+
+  timingSafeEqual: typeof import("node:crypto").timingSafeEqual;
+}
 declare var Crypto: {
   prototype: Crypto;
   new (): Crypto;
 };
-
 declare var crypto: Crypto;
 
 /**

--- a/test/integration/bun-types/fixture/crypto.ts
+++ b/test/integration/bun-types/fixture/crypto.ts
@@ -1,3 +1,37 @@
-import { webcrypto } from "crypto";
+import * as nodeCrypto from "crypto";
+import { expectType } from "./utilities";
 
-webcrypto.CryptoKey;
+nodeCrypto.webcrypto.CryptoKey;
+
+crypto.getRandomValues(new Uint8Array(1));
+
+crypto.subtle.deriveKey(
+  "HMAC",
+  await crypto.subtle.importKey("raw", new TextEncoder().encode("secret"), "HMAC", false, ["deriveKey"]),
+  { name: "HMAC", hash: "SHA-256" },
+  false,
+  ["sign", "verify"],
+);
+
+await crypto.subtle.generateKey("HMAC", false, ["sign", "verify"]);
+
+declare const key: CryptoKey;
+
+crypto.subtle.digest("SHA-256", new TextEncoder().encode("secret"));
+crypto.subtle.exportKey("jwk", key);
+crypto.subtle.importKey("raw", new TextEncoder().encode("secret"), "HMAC", false, ["sign", "verify"]);
+crypto.subtle.encrypt("AES-CBC", key, new TextEncoder().encode("secret"));
+crypto.subtle.decrypt("AES-CBC", key, new TextEncoder().encode("secret"));
+
+expectType(crypto.getRandomValues(new Uint8Array(1))).is<Uint8Array<ArrayBuffer>>();
+expectType(crypto.subtle.digest("SHA-256", new TextEncoder().encode("secret"))).is<Promise<ArrayBuffer>>();
+expectType(crypto.subtle.importKey("raw", new TextEncoder().encode("secret"), "HMAC", false, ["sign", "verify"])).is<
+  Promise<CryptoKey>
+>();
+expectType(crypto.subtle.encrypt("AES-CBC", key, new TextEncoder().encode("secret"))).is<Promise<ArrayBuffer>>();
+expectType(crypto.subtle.decrypt("AES-CBC", key, new TextEncoder().encode("secret"))).is<Promise<ArrayBuffer>>();
+
+expectType(crypto.randomUUID()).is<`${string}-${string}-${string}-${string}-${string}`>();
+expectType(
+  crypto.timingSafeEqual(new TextEncoder().encode("secret"), new TextEncoder().encode("secret")),
+).is<boolean>();

